server/routerlicious/packages/services-client/src/storageUtils.ts

```diff
@@ -18,6 +18,16 @@ import {
     INormalizedWholeSummary,
 } from "./storageContracts";
 
+/**
+ * Options defining conversion of IWholeFlatSummary
+ */
+ export interface ISnapshotConversionOptions {
+    /**
+    * Strip the .app/ path from app tree entries such that they are stored under root.
+    */
+    stripAppPath: boolean;
```

Comment:
I am not sure about adding this as a generic option yet. This needs more thought and plan. The reason I am saying this is because currently, the ISummaryTree uploaded to the driver layer doesn't have any `.app` tree. So, when the summary is downloaded (via getSnapshotTree or downloadSummary), the ISummaryTree shouldn't have the `.app` tree. If these two are different, it can lead to inconsistent behavior. Basically, what is uploaded should be what is downloaded.

Another point worth mentioning here is that we are planning to uploading the protocol tree as part of the upload process - https://github.com/microsoft/FluidFramework/blob/ed7b4b556bef3d5e688271f9b2435d0f3d20a97e/packages/loader/container-loader/src/container.ts#L1309. When this happens, the ISummaryTree that is uploaded will already have an `.app` path and that is what should be returned in both downloadSummary and getSnapshotTree.

@anthony-murphy @vladsud

Reply:
I understand this is needed for https://github.com/microsoft/FluidFramework/pull/9650. For now, maybe the `downloadSummary` function in the whole summary factory should call the `convertWholeFlatSummaryToSnapshotTreeAndBlobs` and then strip off the .app tree?

Reply:
@agarwal-navin the change here is somewhat opposite then maybe what you assumed. We don't want to strip `.app` path (by default it was stripped up to this point). This is services-client util function, and the change is towards making it more generic. Any manipulation around app path should be done in upper layers, it seems.

Reply:
I think we need tenets/objectives that we can use to reason about changes we are making. One of the bigger ones we discussed today is not to have too many unnecessary conversion functions. If this utility function has optional capability NOT to strip `.app` then we don't need another "folding" utility function in document service itself. Keep in mind that all this stays at driver level anyways. This function is consumed only at driver level (and service itself).

We already have precedent for this. See `buildHierarchy` that `ShreddedSummaryDocumentStorageService` is calling. This is parallel implementation just for Shredded variant. Stripping of .app is optional there. I'm not sure why `buildHeirarchy` (that we are changing here) cannot have that capability and be different (other that is misspelled as well).

If we want to stick to "let's not have too many conversions spread out", let's enable these converters so they can offer variants. For your last point, if that is true, we may need this option anyways.

Reply:
That makes sense. As we discussed, one thing you could look at is the `convertSummaryTreeToWholeSummaryTree` function which does the opposite of `convertWholeFlatSummaryToSnapshotTreeAndBlobs / buildHierarchy` - It converts from ISummaryTree -> IWholeSummaryTree during summary upload.
It accepts a `rootNodeName` param which can be ".app" or something else and it builds the tree according to that. Maybe `convertWholeFlatSummaryToSnapshotTreeAndBlobs` should do something similar and accept this string as a param and build the tree based on that string. If the string is ".app" it strips of the corresponding subtree. if its empty, it doesn't strip off anything. That will make the two functions symmetrical.

Reply:
Agreed. Left ".app" as default parameter for 2nd argument, so the function is still in line with existing behavior. Added tests so there is more clarity what actually happens during this conversion.